runtime/src/bank.rs

```diff
@@ -6067,6 +6067,41 @@ impl Bank {
         self.epoch_schedule().get_leader_schedule_epoch(slot)
     }
 
+    /// Returns whether the specified epoch should use the new vote account
+    /// keyed leader schedule
+    pub fn should_use_vote_address_leader_schedule(&self, epoch: Epoch) -> Option<bool> {
+        const NEW_LEADER_SCHEDULE_EPOCH_DELAY: u64 = 1;
+
+        // Cannot determine if an epoch should use the new leader schedule if the
+        // the epoch is too far in the future because we won't know if the feature
+        // will have been activated or not.
+        if self.epoch.saturating_add(NEW_LEADER_SCHEDULE_EPOCH_DELAY) < epoch {
+            return None;
+        }
+
+        let use_vote_address_leader_schedule = self
+            .feature_set
+            .activated_slot(&solana_feature_set::enable_vote_address_leader_schedule::id())
+            .map(|activation_slot| {
+                let activation_epoch = self.epoch_schedule().get_epoch(activation_slot);
+                let effective_epoch = if activation_epoch == 0 {
```

Comment:
don't we need to base this off the activation _slot_ being 0 to accommodate activation in epoch 0

Reply:
Yeah that's a better way to do it